controllers/specialresource.go

```diff
@@ -121,10 +129,39 @@ func ReconcilerSpecialResources(r *SpecialResourceReconciler, req ctrl.Request)
 			}
 		}
 
-		//log = r.Log.WithValues("specialresource", r.parent.Name)
-		log = r.Log.WithName(color.Print(r.parent.Name, color.Green))
+		log = r.Log.WithName(color.Print(r.specialresource.Name, color.Green))
 		log.Info("Reconciling")
 
+		isMarkedToBeDeleted := r.specialresource.GetDeletionTimestamp() != nil
```

Comment:
Can we move this whole block into a function an add it to `finalizers.go` file?

Reply:
e.g. ReconcileFinalizers(r)

Reply:
Yep, good idea